proposals/0002-gep.md

```diff
@@ -0,0 +1,207 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers.
+
+## Motivation
+
+[TBU, what problem we are trying to solve, focus on users, use cases, and all important context]
+
+The Unified API holds all the logic needed to create a request to a provider. This solution allows the clients to be easily
+envoked by a provider pool.
+
+
+### Requirements
+
+- R1: User should be able to specify specific models or select auto mode (selection handled by Glide)
+- R2: Should be a Hertz Client
+- R3: Should take a user request as input
+- R4: Handles all provider specific logic
+- R5: Easily maintained
+
+
+## Design
+
+
+- A Hertz client that handles an incoming request and returns a reponse from that provider.
+- Provider specific clients
+
+### Routes
+```yaml
+routes:
+    chat: api/v1/chat
+    complete: /api/v1/complete #this is also know as generate in some providers
+
+    # want to avoid specifying a provider pool as in the future we may
+    # want to combine strategies e.g User employs a fallback strategy where
```

Comment:
Another thing: fallback strategy and load balancing are very connected topics if you think about them. So I'm thinking of not exposing fallback as a config or something like that anywhere. 

It just should be applied automatically if there is more than one provider in the pool in a way load balancing dictates.

Based on that, fallback will be "absorbed" and the config is simplified for the end users. 

Fallback is not an option if you want to have high availability, so I'm not quite get Portkey, for example, that expose it as config like you choose either load balancing or fallback 😌 

Does that make sense?

Reply:
Yep. That seems reasonable.